Source/Particles/Collision/BackgroundStopping/BackgroundStopping.cpp

```diff
@@ -156,6 +159,9 @@ void BackgroundStopping::doBackgroundStoppingOnElectronsWithinTile (WarpXParIter
             amrex::Real const n_e = n_e_func(x, y, z, t);
             amrex::Real const T_e = T_e_func(x, y, z, t)*PhysConst::kb;
 
+            AMREX_ASSERT(n_e > 0.);
+            AMREX_ASSERT(T_e > 0.);
```

Comment:
```suggestion
            AMREX_ASSERT(n_e > 0._rt);
            AMREX_ASSERT(T_e > 0._rt);
```

Reply:
Thanks for your comments @ax3l. Regarding use of `amrex::ParticleReal`, there is some vagueness about when it should be used. My understanding is that `ParticleReal` is for the large particle data arrays to save space. But for other variables, does it matter? Is there extra cost converting between `float` and `double`? Does using doubles slow down the GPU? There is a lot of inconsistency through out WarpX with Real and ParticleReal.

Reply:
Yes, the idea is to use it for all particle quantities. Where unsafe, we use explicit double instead.

Yes, runtime casts are surprisingly expensing, especially on GPU they are a special instruction.